commit 59792cb0b801854ee41be72d33db9542735df754
Author: Max Kellermann <max.kellermann@gmail.com>
Date:   Wed Jul 27 10:59:48 2022 +0200

    decoder/ffmpeg: wrap FFmpeg include in "extern C"
    
    Commit ebae25d175eb31 added that #include, but forgot to wrap it in
    "extern C", so the linker tried to look up C++ symbols, causing linker
    failure.
    
    Closes https://github.com/MusicPlayerDaemon/MPD/issues/1582

diff --git a/src/decoder/plugins/FfmpegIo.cxx b/src/decoder/plugins/FfmpegIo.cxx
index 2e22d9599102..5b5c8b40e3a0 100644
--- a/src/decoder/plugins/FfmpegIo.cxx
+++ b/src/decoder/plugins/FfmpegIo.cxx
@@ -21,10 +21,13 @@
 #define __STDC_CONSTANT_MACROS
 
 #include "FfmpegIo.hxx"
-#include "libavutil/mem.h"
 #include "../DecoderAPI.hxx"
 #include "input/InputStream.hxx"
 
+extern "C" {
+#include <libavutil/mem.h>
+}
+
 AvioStream::~AvioStream()
 {
 	if (io != nullptr) {
